### DISCUSSION TITLE
Why not use `rbd snap create` to create in-image snapshot for rbd?

### DISCUSSION THREAD

**Discussion by jeffrey4l** [ceph/ceph-csi#4046] [Q&A] [answered]
Based on the code, when creating snapshot , it work as following https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/snapshot.go#L27C6-L68

* create a snapshost for the image
* create a new image from the snapshot
* delete the snapshot

The question is why not just use `rbd snap` which is very fast and less disk usage.

**Comment by Rakshith-R** [comment]
@jeffrey4l refer to https://github.com/ceph/ceph-csi/pull/467#issue-467244968 .

**Reply by jeffrey4l** [reply, marked as answer]
Thanks  , seem the finaly merged pr is [pr#1160](https://github.com/ceph/ceph-csi/pull/1160)
